packages/wonder-blocks-popover/components/popover.js

```diff
@@ -0,0 +1,222 @@
+// @flow
+import * as React from "react";
+import ReactDOM from "react-dom";
+
+import {TooltipPopper} from "@khanacademy/wonder-blocks-tooltip";
+import {maybeGetPortalMountedModalHostElement} from "@khanacademy/wonder-blocks-modal";
+
+import type {AriaProps} from "@khanacademy/wonder-blocks-core";
+import type {
+    Placement,
+    PopperElementProps,
+} from "@khanacademy/wonder-blocks-tooltip";
+
+import PopoverContent from "./popover-content.js";
+import PopoverContentCore from "./popover-content-core.js";
+import PopoverContext from "./popover-context.js";
+import PopoverAnchor from "./popover-anchor.js";
+import PopoverDialog from "./popover-dialog.js";
+
+type Props = {|
+    ...AriaProps,
+
+    /**
+     * The element that triggers the popover. This element will be used to
+     * position the popover. It can be either a Node or a function using the
+     * children-as-function pattern to pass an open function for use anywhere
+     * within children. The latter provides a lot of flexibility in terms of
+     * what actions may trigger the `Popover` to launch the popover dialog.
+     */
+    children: React.Element<any> | (({open: () => void}) => React.Node),
+
+    /**
+     * The content of the popover. You can either use
+     * [PopoverContent](#PopoverContent) with one of the pre-defined variants,
+     * or include your own custom content using
+     * [PopoverContentCore](#PopoverContentCore directly.
+     *
+     * If the popover needs to close itself, the close function provided to this
+     * callback can be called to close the popover.
+     */
+    content:
+        | React.Element<typeof PopoverContent>
+        | React.Element<typeof PopoverContentCore>
+        | (({close: () => void}) =>
+              | React.Element<typeof PopoverContentCore>
+              | React.Element<typeof PopoverContent>),
+
+    /**
+     * Where the popover should try to appear in relation to the trigger element.
+     */
+    placement: Placement,
+
+    /**
+     * When enabled, user can hide the popover content by pressing the `esc` key
+     * or clicking/tapping outside of it.
+     */
+    dismissEnabled?: boolean,
```

Comment:
This is not working yet. This will be part of PR 4/4. Same for the `id` and `testId` props.

Reply:
Didn't add this, b/c I already added this functionality in PR 4/4 but next time I'll make sure to do it. Thanks for the suggestion!